File: result/translate/chatgpt/java2python/python/FIND_LARGEST_PRIME_FACTOR_NUMBER.py
def f_gold(n):
    maxPrime = -1
    while n % 2 == 0:
        maxPrime = 2
        n >>= 1

    i = 3
    while i <= int(n**0.5):
        while n % i == 0:
            maxPrime = i
            n = n // i
        i += 2

    if n > 2:
        maxPrime = n

    return maxPrime

File: result/translate/transcoder/python2java/java/FIND_LARGEST_PRIME_FACTOR_NUMBER.java
public static int f_gold ( int n ) {
  int maxPrime = - 1 ;
  while ( n % 2 == 0 ) {
    maxPrime = 2 ;
    n >>= 1 ;
  }
  for ( int i = 3 ;
  i <= Integer . MAX_VALUE ;
  i += 2 ) {
    while ( n % i == 0 ) {
      maxPrime = i ;
      n = n / i ;
    }
  }
  if ( n > 2 ) maxPrime = n ;
  return ( int ) maxPrime ;
}
